### ISSUE TITLE
System.IO.FileNotFoundException: 'Could not load file or assembly 'Microsoft.Azure.Devices.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'

### ISSUE THREAD

**Issue by VadimUmojo** [Azure/azure-iot-sdk-csharp#94]
Referencing Microsoft.Azure.Devices.Client 1.2.9 in .NET Standard 1.4 library:

System.IO.FileNotFoundException: 'Could not load file or assembly 'Microsoft.Azure.Devices.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'

**Comment by jasmineymlo** [comment]
Are you saying you are building a .net standard 1.4 library referencing to Microsoft.Azure.Devices.Client 1.2.9?  Was it at run time that you got the error?

**Comment by VadimUmojo** [comment]
Runtime exception

**Comment by jasmineymlo** [comment]
I am assuming that you referenced Microsoft.Azure.Devices.Client from your .NetStandard 1.4 library.  Then create another application project within the solution and referenced to the .NetStandard 1.4 dll directly.

The application project would not be able to find reference to the installed nuget packages as they are in a different bin folder.  You can add reference to Microsoft.Azure.Devices.Client in your application project as well to work around this.  Hope this helps.

**Comment by jasmineymlo** [comment]
Please let us know if you have further issue.  I am closing this thread as of now...

**Comment by DazFahy** [comment]
Hi

I've got the same issue but with creating a .NetStandard 1.4 library which references Microsoft.Azure.Devices.Client 1.2.13.

I'm trying to reference my library as a nuget package in a UWP IoT Background Application and get the same runtime exception. The above workaround does not work, if I include the Microsoft.Azure.Devices.Client 1.2.13 in the UWP IoT Background Application I still get the same error.

The only solution I've found is to create a UWP Class Library and physically copy the files from the .NetStandard 1.4 library into it and reference UWP Class Library in the UWP IoT Background Application. However this now means I've got two copies of the code as a UWP Class Library can only be referenced by a UWP Application.
